internal/tools/aws/database_migration.go

```diff
@@ -0,0 +1,132 @@
+		return "", err
+	}
+
+	_, err = d.awsClient.Service().ec2.AuthorizeSecurityGroupIngress(&ec2.AuthorizeSecurityGroupIngressInput{
+		GroupId: masterInstanceSG.GroupId,
+		IpPermissions: []*ec2.IpPermission{
+			{
+				FromPort:   aws.Int64(3306),
+				IpProtocol: aws.String("tcp"),
+				ToPort:     aws.Int64(3306),
+				UserIdGroupPairs: []*ec2.UserIdGroupPair{
+					{
+						Description: aws.String("Ingress Traffic from other RDS instance"),
+						GroupId:     slaveInstanceSG.GroupId,
+					},
+				},
+			},
+		},
+	})
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return "", err
+	}
+
+	logger.WithField("migration-installation-id", d.masterInstallationID).Info("database migration setup completed")
+
+	return model.DatabaseMigrationStatusSetupComplete, nil
+}
+
+// Teardown removes access from one RDS database to another and rollback any previous database configuration.
+func (d *RDSDatabaseMigration) Teardown(logger log.FieldLogger) (string, error) {
+	masterInstanceSG, err := d.describeDBInstanceSecurityGroup(RDSMasterInstanceID(d.masterInstallationID))
+	if err != nil {
+		return "", err
+	}
+
+	slaveInstanceSG, err := d.describeDBInstanceSecurityGroup(RDSMigrationInstanceID(d.slaveInstallationID))
+	if err != nil {
+		return "", err
+	}
+
+	_, err = d.awsClient.Service().ec2.RevokeSecurityGroupIngress(&ec2.RevokeSecurityGroupIngressInput{
+		GroupId: masterInstanceSG.GroupId,
+		IpPermissions: []*ec2.IpPermission{
+			{
+				FromPort:   aws.Int64(3306),
+				IpProtocol: aws.String("tcp"),
+				ToPort:     aws.Int64(3306),
+				UserIdGroupPairs: []*ec2.UserIdGroupPair{
+					{
+						GroupId: slaveInstanceSG.GroupId,
+					},
+				},
+			},
+		},
+	})
+	if err != nil && !strings.Contains(err.Error(), "rule does not exist") {
+		return "", err
+	}
+
+	logger.WithField("migration-installation-id", d.masterInstallationID).Info("database migration teardown completed")
+
+	return model.DatabaseMigrationStatusTeardownComplete, nil
+}
+
+// Replicate starts the process for replicating an master RDS database. This method must return an resplication status or an error.
+func (d *RDSDatabaseMigration) Replicate(logger log.FieldLogger) (string, error) {
+	return "", errors.New("not implemented")
+}
+
+func (d *RDSDatabaseMigration) describeDBInstanceSecurityGroup(instanceID string) (*ec2.SecurityGroup, error) {
+	output, err := d.awsClient.Service().rds.DescribeDBInstances(&rds.DescribeDBInstancesInput{
+		DBInstanceIdentifier: aws.String(instanceID),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, instance := range output.DBInstances {
+		for _, vpcSG := range instance.VpcSecurityGroups {
+			sgOutput, err := d.awsClient.Service().ec2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+				GroupIds: []*string{vpcSG.VpcSecurityGroupId},
+			})
+			if err != nil {
+				return nil, err
+			}
+			if len(sgOutput.SecurityGroups) == 1 && strings.Contains(*sgOutput.SecurityGroups[0].GroupName, "-db-sg") {
```

Comment:
I can't seem to find where the security group name's naming scheme is forced to include `-db-sg` anywhere, I suppose at security group creation time? Can you help me find that? If it's not enforced and is just by convention, I think the convention should be encoded so that it is enforced, and this hardcoded string `"-db-sg"` should be updated to use a constant (presumably the constant would also be useful at convention-enforcement time).

Reply:
It's also fine if this :point_up: is going to be in the next PR :)

Reply:
This is probably in Terraform:
```
mattermost-cloud-dev-provisioning-1024000-db-sg
```

Reply:
Hmm, if it's from Terraform I guess I don't feel too strongly about it one way or the other. A comment on that line saying where the value comes from might be good, but it's fine even without it.

Reply:
That's a very good point. I can try to find some other metadata but it will be from Terraform.